evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -353,86 +362,93 @@ def test_simple_imputer_with_none():
-            "cat with None": ["a", "b", "a", None],
-            "natural language col": ["free-form text", "will", "be imputed", None],
+            "Categorical": ["a", "b", "a", na_type],
+            "NaturalLanguage": ["free-form text", "will", "be imputed", na_type],
         }
     )
+    X = X[[data_type]]
     y = pd.Series([0, 0, 1, 0, 1])
-    X.ww.init(
-        logical_types={
-            "cat with None": "categorical",
-            "natural language col": "NaturalLanguage",
-        }
-    )
+
+    X.ww.init(logical_types={data_type: data_type})
     imputer = SimpleImputer(impute_strategy="constant", fill_value="placeholder")
     imputer.fit(X, y)
     transformed = imputer.transform(X, y)
     expected = pd.DataFrame(
         {
-            "cat with None": pd.Series(
-                ["a", "b", "a", "placeholder"], dtype="category"
-            ),
-            "natural language col": pd.Series(
-                ["free-form text", "will", "be imputed", "placeholder"], dtype="string"
+            "Categorical": pd.Series(["a", "b", "a", "placeholder"], dtype="category"),
+            "NaturalLanguage": pd.Series(
+                ["free-form text", "will", "be imputed", pd.NA], dtype="string"
             ),
         }
-    )
+    )[[data_type]]
     assert_frame_equal(expected, transformed, check_dtype=False)
 
 
 @pytest.mark.parametrize(
-    "X_df",
+    "data",
     [
-        pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
-        pd.DataFrame(pd.Series([1.0, 2.0, 3.0], dtype="float")),
-        pd.DataFrame(pd.Series(["a", "b", "a"], dtype="category")),
-        pd.DataFrame(pd.Series([True, False, True], dtype=bool)),
-        pd.DataFrame(
-            pd.Series(
-                ["this will be a natural language column because length", "yay", "hay"],
-                dtype="string",
-            )
-        ),
+        "integer_data",
+        "float_data",
+        "categorical_data",
+        "boolean_data",
     ],
 )
-@pytest.mark.parametrize("has_nan", [True, False])
+@pytest.mark.parametrize(
+    "logical_type", ["Integer", "Double", "Categorical", "NaturalLanguage", "Boolean"]
+)
+@pytest.mark.parametrize("has_nan", ["has_nan", "no_nans"])
 @pytest.mark.parametrize("impute_strategy", ["mean", "median"])
 def test_simple_imputer_woodwork_custom_overrides_returned_by_components(
-    X_df, has_nan, impute_strategy
+    data, logical_type, has_nan, impute_strategy
 ):
+    X_df = {
+        "integer_data": get_int_df,
+        "float_data": get_float_df,
+        "categorical_data": get_category_df,
+        "boolean_data": get_bool_df,
+    }[data]()
+    logical_type = {
+        "Integer": Integer,
+        "Double": Double,
+        "Categorical": Categorical,
+        "NaturalLanguage": NaturalLanguage,
+        "Boolean": Boolean,
+    }[logical_type]
     y = pd.Series([1, 2, 1])
-    override_types = [Integer, Double, Categorical, NaturalLanguage, Boolean]
-    for logical_type in override_types:
```

Comment:
Refactored this test so that all the parameters are listed in the test failure sections and it's easier to understand what's failing, specifically which override type.